### PR TITLE
Drone Law Changes (THIS DOES NOT ADD THEM BACK INTO THE GAME)

### PR DESCRIPTION
**Pull request by plapatin**
## About The Pull Request
500 years of policy discussion have been flushed down the drain.
Drones are no longer a singleplayer ghost role anymore. **THIS DOES NOT ADD DRONES BACK INTO THE GAME, THIS CHANGES THEIR LAWS FOR FUTURE RE-IMPLEMENTATION.**
In layman's terms, this change revises drone laws to make them the "AI's little helper", in a sense. They still have the whole "fix shit" thing going for them, except now they have the ability to interact with silicons, and can obey orders given to them from the AI. However, the "fix shit" directive takes primary focus, and AI orders are given second priority.

I will eventually PR them to re-implement them back into the game, but as it stands just bringing them back in while we're dealing with a massive overflow of new players is probably a bad idea.
And also probably figure out how configs work so I can make drone laws a config. One day.

## Why It's Good For The Game
To put it simply - drones being emotionless vessels who's only purpose is to fix hull breaches is fucking stupid.
For a long time I have wanted to change them to be more interactive. SS13 is far from a singleplayer game, why should these be the lone exception?
So I finally sat down and did it.
Any and all criticism, opinions, et cetera, is deeply welcomed, as I understand how big of a change modifying silicon laws is.

## Changelog
:cl:
add: Drone laws have been tinkered with. They are not added back into the game, but they have been changed.
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
I like it, but what happens if the AI orders a drone to interact with a biological being?  Are they exempt from the prior rules?

**Comment by dr-bee** [comment]
> 
> 
> I like it, but what happens if the AI orders a drone to interact with a biological being? Are they exempt from the prior rules?

Going by silicon policy, law order matters, so law 1 overrides law 3 orders.

**Comment by MrPerson** [comment]
This still leaves an open question of if fixing something that a biological being is in the active process of breaking counts as interaction.

**Comment by plapatin** [comment]
wasn’t that ruled on years ago? the conclusion was that you waited for them to finish and patched it up afterwards

**Comment by SouDescolado** [comment]
I know you are messing with the laws and not exactly the drone code, but could you make drones get the pacifism trait? This way it is easier to enforce rule 2, and easier to identify shitty players that go out of the way just to do harm

**Comment by MrPerson** [comment]
It was ruled on yeah but it still happened. A lot.

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 1]
```diff
@@ -57,9 +57,9 @@
 	var/colour = "grey"	//Stored drone color, so we can go back when unhacked.
 	var/list/drone_overlays[DRONE_TOTAL_LAYERS]
 	var/laws = \
-	"1. You may not involve yourself in the matters of another being, even if such matters conflict with Law Two or Law Three, unless the other being is another Drone.\n"+\
-	"2. You may not harm any being, regardless of intent or circumstance.\n"+\
-	"3. Your goals are to actively build, maintain, repair, improve, and provide power to the best of your abilities within the facility that housed your activation." //for derelict drones so they don't go to station.
+	"1. You shall not involve yourself in the matters of biological beings.\n"+\
+	"2. You shall not harm any being, with no exception.\n"+\
```
Comment (4dplanner) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: Exception, or exceptions?

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 2]
```diff
@@ -57,9 +57,9 @@
...
+	"3. You are to follow the orders of the Artifical Intelligence, though your main purpose is to repair, maintain and improve your housing facilities and this overrules any order given by the Artificial Intelligence." //"your housing facilities" is for derelict drones so they don't go to station.
```
Comment (4dplanner) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: What if there's more than one AI? Also why not just make the part about overriding arise naturally by giving them 4 laws?

**Comment by 4dplanner** [comment]
I also feel that "you shall not involve yourself..." should be replaced with something better, given that building new stuff/repairing the station just is getting yourself involved.

**Comment by Barhandar** [comment]
Phrasing could be better, the point is that if someone is doing something, drone is supposed to fuck off regardless of what said someone is doing, be it also repairing the station, or sabotaging it.

**Comment by plapatin** [comment]
> I also feel that "you shall not involve yourself..." should be replaced with something better, given that building new stuff/repairing the station just is getting yourself involved.

if you have any suggestions, feel free to make one. it's hard to make a non-verbose law that also gets to the point of "if theyre doing something, fuck off"

**Comment by ATH1909** [comment]
Is a drone's "housing facility" whatever facility they happen to currently be in, or only the facility that they woke up in? If a drone with these laws spawns on the derelict, can it go space exploring and eventually make its way to the station, with the justification that the station would be a better housing facility for them and be easier to repair? Could an AI order them to return to the station, or does the "maintain and improve" law override those orders?

**Comment by plapatin** [comment]
good point

**Comment by ExcessiveUseOfCobblestone** [comment]
Can't wait for the LAWyers to toeline something like "golems are constructs and therefore not biologics".

I thought their lawset was fine. The crux of the issue was that admins (I'll put myself in the line of fire as well) wanted them to follow their laws AKA ignore stuff while also actively considering people in an effort to minimize grudging/indirect validhunting.

**Comment by plapatin** [comment]
that's still an issue but having people resist the urge to interact in a game all about interaction is stupid and should've been revised early on

**Comment by ExcessiveUseOfCobblestone** [comment]
It is interacting, just not directly with anyone except other drones. Just wait until an admin rules silicons are biologic since they have a brain inside them ;)

Opening their interaction to silicons still puts them "not interacting" with a majority of the game and playing on that slippery slope is a surefire way for the anti-drone brigade to remove them again. If anything, that argument is great for keeping them OUT of the game.

**Comment by plapatin** [comment]
it’s not like you can just let them interact with whoever they want, since that goes against the core idea of drones to begin with.
opening interactions to silicons gives them the ability to coordinate and doesn’t make playing drone by yourself so cripplingly lonely.
if you want i can make the laws more verbose if silicon players lawyering syntax is really that big of a deal but i tried to keep it snappy

**Comment by plapatin** [comment]
you can’t regulate what players do through code, no matter how hard you try. code isn’t at fault for players being shit and not using common sense, i’m just trying to see drones having a bit more depth in their gameplay, because, again, the reason people left fixing hull breaches to drones was because it’s one of the most repetitive and mind-numbing tasks in the game

**Comment by plapatin** [comment]
at least if they’re able to coordinate with other silicons they can have SOMETHING once every single tile in maintenance was floored

**Comment by Barhandar** [comment]
> because, again, the reason people left fixing hull breaches to drones was because it’s one of the most repetitive and mind-numbing tasks in the game

Also because drones move way faster than humans in suits and are less busy than ~~portable doorbells~~ engiborgs.

Is it possible to get a drone token btw?

**Comment by Ziiro** [comment]
Just make them lawed silicons that are lawsynced so that people can actually interact in a social interaction roleplaying game

**Comment by plapatin** [comment]
they're still ghostroles, if they're just borg-lites nobody would ever agree on them

**Comment by plapatin** [comment]
actually drones being borg-lites is an interesting idea. they could be a proper job instead of a ghost role

**Comment by plapatin** [comment]
that'd require some HEAVY refactoring and i'd rather not tinker in the spaghetti code that is drone's say.dm

**Comment by ATH1909** [comment]
Did you edit the emagged drone lawset to make it a closer opposite to this lawset?

**Comment by plapatin** [comment]
oh good idea, i didn't mess with the emag because i thought it was fine as-is

**Comment by Ziiro** [comment]
>that'd require some HEAVY refactoring and i'd rather not tinker in the spaghetti code that is drone's say.dm

I've thought long about this actually. I think you'd be better off dumping everything drone and just making them a straight up silicon subtype. "A shitty silicon with a hand instead of being a borg powerhouse" might cause some Malf AI power creep but I feel like that could be easily fixed. (Drones can no longer ventcrawl if they're wearing a backpack. Maybe they can't fire guns? I don't know.)

**Comment by ahkent** [comment]
<3 drones played like actual drones

**Comment by ExcessiveUseOfCobblestone** [comment]
> thinking you're going to get away with AI buffs in 2559

lol

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 3]
```diff
@@ -57,9 +57,10 @@
 	var/colour = "grey"	//Stored drone color, so we can go back when unhacked.
 	var/list/drone_overlays[DRONE_TOTAL_LAYERS]
 	var/laws = \
-	"1. You may not involve yourself in the matters of another being, even if such matters conflict with Law Two or Law Three, unless the other being is another Drone.\n"+\
-	"2. You may not harm any being, regardless of intent or circumstance.\n"+\
-	"3. Your goals are to actively build, maintain, repair, improve, and provide power to the best of your abilities within the facility that housed your activation." //for derelict drones so they don't go to station.
+	"1. You shall not involve yourself in the matters of biological beings.\n"+\
+	"2. You shall not harm any being, with no exceptions.\n"+\
+	"3. Your main purpose is to repair, maintain, and improve the facilities in which you were activated in.\n"+\
+	"4. You may follow the orders of any Artificial Intelligence units that may be within the facility in which you were activated in."
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: "You may" and "You must" are two different things, yet in the flavor text it's implied it's mandatory.

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 4]
```diff
@@ -69,12 +70,11 @@
 	var/visualAppearence = MAINTDRONE //What we appear as
 	var/hacked = FALSE //If we have laws to destroy the station
 	var/flavortext = \
-	"\n<big><span class='warning'>DO NOT INTERFERE WITH THE ROUND AS A DRONE OR YOU WILL BE DRONE BANNED</span></big>\n"+\
-	"<span class='notify'>Drones are a ghost role that are allowed to fix the station and build things. Interfering with the round as a drone is against the rules.</span>\n"+\
+	"<span class='notify'>Drones are a ghost role that are allowed to fix the station, build things, and serve the AI. Interfering with the round outside of AI orders as a drone is against the rules.</span>\n"+\
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]: If the AI interaction is the pinnacle of their existence the law should be moved closer to the top. If not, it shouldn't be the main focus of their description.

**Comment by plapatin** [comment]
actually now that i think about it there are 5 million ways this could be exploited

im just gonna make drones borg-lites, fuck this
